autoscale/src/main/java/com/sequenceiq/periscope/model/MandatoryScalingAdjustmentParameters.java

```diff
@@ -0,0 +1,46 @@
+package com.sequenceiq.periscope.model;
+
+import java.util.Optional;
+
+public class MandatoryScalingAdjustmentParameters {
+
+    private Optional<Integer> mandatoryUpscaleAdjustment;
+
+    private Optional<Integer> mandatoryDownscaleAdjustment;
+
+    private Optional<Integer> mandatoryStopStartUpscaleAdjustment;
+
+    private Optional<Integer> mandatoryStopStartDownscaleAdjustment;
```

Comment:
I am a little bit confused based on the namings and types here. So here we have the class `MandatoryScalingAdjusmentParameters` class which indicates that all of these are mandatory. Also all of the fields have the `mandatory` prefix in name, but only 2 is filled from 4 in case of regular, and in case of stopstart scaling. 

I would suggest the following refactor:
- case of regular scale, we can have `RegularScalingAdjusment` with 2 both integer fields (upscaleAdjustment, downscaleAdjusment), you can even indicate with `@NotNull` that these are required
- case of stopstart, we can have `StopStartScalingAdjustment` with two other integer fields

as I see there is no case when all of these are filled in, so mandatory prefixes are quite misleading

Reply:
Thank you!